jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/frames/HeadersFrame.java

```diff
@@ -41,6 +41,11 @@ public HeadersFrame(MetaData metaData, PriorityFrame priority, boolean endStream
         this(0, metaData, priority, endStream);
     }
 
+    public HeadersFrame(int streamId, PriorityFrame priority, HeadersFrame other)
+    {
+        this(streamId, other.getMetaData(), priority, other.isEndStream());
+    }
+
```

Comment:
If this constructor is only used by the implementation, remove it so it's not expose to applications.

Reply:
Sure.